src/Query/Builder.php

```diff
@@ -538,9 +540,26 @@ public function generateCacheKey()
         return md5(serialize(array_values($key)));
     }
 
-    /** @inheritdoc */
-    public function aggregate($function, $columns = [])
+    /** @return ($function is null ? AggregationBuilder : mixed) */
+    public function aggregate($function = null, $columns = ['*'])
     {
+        if ($function === null) {
+            if (! trait_exists(FluentFactoryTrait::class)) {
+                // This error will be unreachable when the mongodb/builder package will be merged into mongodb/mongodb
+                throw new BadMethodCallException('Aggregation builder requires package mongodb/builder 0.2+');
+            }
+
+            if ($columns !== [] && $columns !== ['*']) {
```

Comment:
Is there any reason to allow `[]` for `$columns`? IIRC, the previous signature changed the default value to an empty array, but you're preserving `['*']` now. Do you expect users to explicitly call `aggregate(null, [])`? If not, I'd stick to just `['*']` here.